rts/motoko-rts/src/persistence.rs

```diff
@@ -155,16 +155,16 @@ pub(crate) unsafe fn stable_actor_location() -> *mut Value {
 /// Used for upgrading to an actor with additional stable fields.
 #[no_mangle]
 #[cfg(feature = "ic")]
-pub unsafe extern "C" fn contains_field(actor: Value, field_hash: u32) -> bool {
+pub unsafe extern "C" fn contains_field(actor: Value, field_hash: usize) -> bool {
```

Comment:
field_hash is actually an extended 32-bit hash, correct? We don't store the top half of the word? I ask because Candid uses 32-bit hashes and I expect they need to agree with Motoko's field hashes. Indeed, we could probably halve the size of the record field descriptors blob if you are storing the hashes as 64-bit.

Reply:
Absolutely, the hashes stay 32-bit values and we could compact the hash blob.

If okay, I put it on my `u32` downgrade backlog.

Reply:
I wonder if we'll regret not compacting the type descriptors now - since it would require a graph copy migration later. But perhaps just remaining with non-compact type descriptors fine. I guess one could even use the slack space to refer to a string table for precise field names in future....

Totally up to you - just wanted to point out the graph migration issue. The other u32/usize changes can indeed be done later sans migration.

Reply:
I agree, now we can do this relatively easily. I'll prepare the change and give an update when ready.